src/main/java/com/ecommerce/ECommerceApp/model/Order.java

```diff
@@ -12,7 +13,7 @@ public class Order {
     private int orderID;
     private LocalDateTime orderDate;
     private int userID;
-    private float totalAmount;
+    private BigDecimal totalAmount;
```

Comment:
Replaced float with BigDecimal for monetary values to prevent rounding issues in financial calculations, ensuring precision and reliability. As Martin Fowler discusses in ["Technical Debt"](https://martinfowler.com/bliki/TechnicalDebt.html), addressing data accuracy early on reduces technical debt and avoids potential issues down the line.